### ISSUE TITLE
Improve visibility into slow deployment directory copies

### ISSUE THREAD

**Issue by madkinsz** [PrefectHQ/prefect#8192]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Deployment copies can be very slow if it includes many directories or files. This can be surprising to users (e.g. #8178). In #8075 I added some logs to give additional visibility into where the flow run is stalled. However, there is still not enough clarity into what is happening during 5+ minute delays.

### Describe the proposed behavior

Not all of these are necessary and some may not be feasible:

1. Add a log for each file or directory copied
2. Add a warning log if the copy takes longer than N seconds that explains how to limit the scope of copies
3. Add a warning log if copy of a large file takes place
4. Improve documentation on working directories for deployments

**Comment by richard-alexander** [comment]
Thank you Michael!  :-)

**Comment by tjgalvin** [comment]
This one has bitten me before as well. These logs would be very useful :)

**Comment by madkinsz** [comment]
@yusuf-jkhan1 are there any hidden files? Perhaps a `.git` directory that's not being ignored?

Contributions are welcome here fyi!

**Comment by yusuf-jkhan1** [comment]
Yeah no .git directory. I was on a call with a sales engineer and showed them too. Going to be on another call with pro serv hopefully they can help.
